File: 0062-unique-paths/0062-unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        
        vector<vector<int>> arr(m,vector<int>(n,0));
        
        for(int i=0;i<n;i++){
            arr[m-1][i]=1;
        }
        
        for(int j=0;j<m;j++){
            arr[j][n-1]=1;
        }
       
        
        for(int i=m-2;i>=0;i--){
            for(int j = n-2;j>=0;j--){
                arr[i][j] = arr[i][j+1] + arr[i+1][j];
            }
        }
        
        return arr[0][0];
        
        
    }
};

File: 0062-unique-paths/0062-unique-paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[][] arr = new int[m+1][n+1];
        
        for(int i=0;i<m;i++)arr[i][0]=0;
        
        for(int i=0;i<n;i++)arr[0][i]=0;
        arr[1][1]=1;
        for(int i=1;i<=m;i++){
            for(int j=1;j<=n;j++){
                arr[i][j] = arr[i][j]+arr[i-1][j]+arr[i][j-1];
            }
        }
        return arr[m][n];
        
    }
}